src/runtime/propensity-server-test.js

```diff
@@ -296,4 +334,81 @@ describes.realWin('PropensityServer', {}, env => {
           expect(() => {throw reason;}).to.throw(/Invalid request/);
         });
   });
+
+  it('should listen for events from event manager', function*() {
+    eventManagerMock.verify();
+  });
+
+  it('should not send SwG events to Propensity Service', () => {
+    //stub to ensure the server received a request to log
+    let receivedType = null;
+    let receivedContext = null;
+    sandbox.stub(Xhr.prototype, 'fetch', url => {
+      const event = getPropensityEventFromUrl(url);
+      receivedType = event.name;
+      receivedContext = event.data;
+      return Promise.reject('Server down');
+    });
+
+    //no experiment set & not activated
+    defaultEvent.eventOriginator = EventOriginator.SWG_CLIENT;
+    propensityServer.handleClientEvent_(defaultEvent);
+    expect(receivedType).to.be.null;
+    expect(receivedContext).to.be.null;
+
+    defaultEvent.eventOriginator = EventOriginator.AMP_CLIENT;
+    propensityServer.handleClientEvent_(defaultEvent);
+    expect(receivedType).to.be.null;
+    expect(receivedContext).to.be.null;
+
+    //activated but no experiment
+    propensityServer.enableLoggingGoogleEvents();
+    propensityServer.handleClientEvent_(defaultEvent);
+    expect(receivedType).to.be.null;
+    expect(receivedContext).to.be.null;
+
+    defaultEvent.eventOriginator = EventOriginator.SWG_CLIENT;
+    propensityServer.handleClientEvent_(defaultEvent);
+    expect(receivedType).to.be.null;
+    expect(receivedContext).to.be.null;
+
+    //experiment but not activated
+    setExperiment(win, ExperimentFlags.LOG_SWG_TO_PROPENSITY, true);
+    eventManagerMock.expects('registerEventListener').once();
+    propensityServer = new PropensityServer(win, pubId, eventManager);
+    propensityServer.handleClientEvent_(defaultEvent);
+    expect(receivedType).to.be.null;
+    expect(receivedContext).to.be.null;
+    setExperiment(win, ExperimentFlags.LOG_SWG_TO_PROPENSITY, false);
+  });
+
+  it('should send SwG events to the Propensity Service', () => {
+    //stub to ensure the server received a request to log
+    let receivedType = null;
+    let receivedContext = null;
+    sandbox.stub(Xhr.prototype, 'fetch', url => {
+      const event = getPropensityEventFromUrl(url);
+      receivedType = event.name;
+      receivedContext = event.data;
+      return Promise.reject('Server down');
+    });
+
+    setExperiment(win, ExperimentFlags.LOG_SWG_TO_PROPENSITY, true);
+    eventManagerMock.expects('registerEventListener').once();
+    propensityServer = new PropensityServer(win, pubId, eventManager);
+
+    //both experiment and enable: ensure it actually logs
+    propensityServer.enableLoggingGoogleEvents();
+    propensityServer.handleClientEvent_(defaultEvent);
```

Comment:
should not have to call internal private methods to do this test.

Reply:
Done?